include/qb/qbloop.h

```diff
@@ -36,6 +36,37 @@ extern "C" {
  *
  * Main loop manages timers, jobs and polling sockets.
  *
+ * The concept of priorities, as implemented in this native event loop, does
+ * not follow the absolute terms as traditionally established in the field
+ * (see, e.g., IRQ analogy), but rather approaches it with a probabilistic
+ * (proportional) method.  There are hence no guarantees as to the order
+ * the callbacks will be invoked in when multiple concurrent events are
+ * awaiting such handling.  This advisory effect of priority assignment can
+ * be seen as a trade-off yielding implicit mitigation for deadlock-prone
+ * priority arrangements in return.
+ *
+ * One practical application for this module of libqb is in combination
+ * with IPC servers based on qbipcs.h published one, whereby the
+ * #qb_ipcs_poll_handlers structure maps nicely to the control functions
+ * published here.  At that occasion, please note the original alignment of
+ * #QB_IPCS_RATE_FAST etc. symbolic names with said advisory effect of the
+ * priorities here, since it gets semantically shifted once the implementation
+ * strictly abiding them is used instead of this native event loop.
+ *
+ * One such alternative with strict approach to the event priorities is
+ * GLib's event loop over poll'able sources.  Speaking of analogies, it can be
+ * compared to real-time scheduling of the processes per the fixed priorities
+ * vs. established "niceness" property only adjusting proportions of their
+ * interleaving, as is represented with this libqb's native loop management.
+ *
+ * Word of caution: developing the application using the native loop
+ * implementation first and exchanging it for the other based on this opposite
+ * paradigm later on may cause said unexpected blocking situations to emerge,
+ * since the former just used to provide a false sense of correctness there.
+ * The inverse direction is naturally smoother, since the code was likely
+ * well-balanced regarding prioritites already, but less overall performance
+ * can be expected if they were downright fine-tuned for the use case before.
```

Comment:
While this is fairly simple to read, it adds virtually nothing of use to a developer.

Overall, virtually none of this is documentation; most of it commentary, discussion, and comparison with other designs. As such, it is much more like a blog post. Creating a wiki page for such lengthy discussions and then linking that from the API documentation is a very reasonable approach. The API can have the bare minimum to document what the API actually does, and discussion of the implications of the design can move at a separate pace, outside the code repository.